src/utils/get_size/mod.rs

```diff
@@ -5,6 +5,13 @@ use cid::Cid;
 use derive_more::{From, Into};
 use get_size2::GetSize;
 
+pub fn vec_alike_get_size<V, T>(slice: &V) -> usize
+where
+    V: AsRef<[T]>,
+{
+    std::mem::size_of_val(slice.as_ref())
+}
+
```

Comment:
Can you help me understand the correctness of this? This seems to work at most with shallow collections, but something like this will totally not work:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=a8109b6ba152d250b00a46a026ccb91d